framework/src/ble_manager/ble_manager_autoconnect.c

```diff
@@ -0,0 +1,249 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <sched.h>
+#include <pthread.h>
+#include <fcntl.h>
+#include <mqueue.h>
+#include <string.h>
+#include "ble_manager_common.h"
+#include "ble_manager_autoconnect.h"
+#include "ble_manager_log.h"
+
+static ble_client_ctx *g_ctx = NULL;
+
+static void ble_auto_scan_state_changed_cb(ble_scan_state_e scan_state)
+{
+	int status = 0;
+	char buf[BLE_MQ_SIZE] = { 0, };
+	
+	if (scan_state == BLE_SCAN_STARTED) {
+		buf[0] = (char)BLE_AUTOCON_EVT_SCAN_START;
+	} else if (scan_state == BLE_SCAN_STOPPED) {
+		buf[0] = (char)BLE_AUTOCON_EVT_SCAN_STOP;
+	} else {
+		return;
+	}
+	
+	status = mq_send(g_ctx->send_mqfd, buf, BLE_MQ_SIZE, 0);
+	if (status < 0) {
+		BLE_LOG_ERROR("[BLEMGR] fail to send mqueue[%d]\n", status);
+	}
+	return;
+}
+
+static void ble_auto_device_scanned_cb(ble_scanned_device *scanned_device)
+{
+	if (g_ctx == NULL) {
+		return;
+	}
+
+	int status = 0;
+	char buf[BLE_MQ_SIZE] = { 0, };
+
+	if (scanned_device->adv_type == BLE_ADV_TYPE_DIRECT) {
+		if (memcmp(g_ctx->info.addr.mac, scanned_device->conn_info.addr.mac, BLE_BD_ADDR_MAX_LEN) == 0) {
+			buf[0] = (char)BLE_AUTOCON_EVT_MAC_SCANNED;
+			status = mq_send(g_ctx->send_mqfd, buf, BLE_MQ_SIZE, 0);
+			if (status < 0) {
+				BLE_LOG_ERROR("[BLEMGR] fail to send mqueue[%d]\n", status);
```

Comment:
How about printing the error number also?
Then it can be helpful to debug when fails.

Reply:
Ok, I will add errno in all mq commands